### PR TITLE
ME-1841: add policy related resources

### PR DESCRIPTION
**Pull request by waltzofpearls**
- add resource `border0_policy`
- add resource `border0_policy_attachment`
- add data source `border0_policy_document`
- add policy resources to `examples/development/main.tf`
- add unit tests for connector and connector token
- add unit tests for policy, policy attachment and policy document
- generate mock for border0 api client
- update terraform docs with the new resources
- use the latest border0-go that supports policy management

### CODE REVIEW

File: border0/resource_policy.go
Comment (waltzofpearls) [inline border0/resource_policy.go:1]: This file and `resource_policy_attachment.go` are the main changes of this PR. Others are unit tests, docs and examples changes.

File: border0/resource_policy_attachment.go
Comment (waltzofpearls) [inline border0/resource_policy_attachment.go:1]: This file and `resource_policy.go` are the main changes of this PR. Others are unit tests, docs and examples changes.

File: border0/resource_connector_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		// terraform aplly (update + read + read)
+		clientMock.EXPECT().UpdateConnector(matchContext, &updateInputOutput).Return(&updateInputOutput, nil).Call,
+		clientMock.EXPECT().Connector(matchContext, "unit-test-id-1").Return(&updateInputOutput, nil).Call,
+		clientMock.EXPECT().Connector(matchContext, "unit-test-id-1").Return(&updateInputOutput, nil).Call,
+
+		// terraform import (read)
+		clientMock.EXPECT().Connector(matchContext, "unit-test-id-1").Return(&updateInputOutput, nil).Call,
+
+		// terraform destroy (delete)
+		clientMock.EXPECT().DeleteConnector(matchContext, "unit-test-id-1").Return(nil).Call,
```
Comment (adrianosela) [inline border0/resource_connector_test.go:64]: no read after delete?

File: border0/resource_connector_test.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	border0client "github.com/borderzero/border0-go/client"
+	"github.com/borderzero/terraform-provider-border0/mocks"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+var initialConnectorConfig = `
+resource "border0_connector" "unit_test" {
+  name = "unit-test-connector-1"
+  description = "connector created from terraform unit test"
+}
```
Comment (adrianosela) [inline border0/resource_connector_test.go:15]: in the future would be nice to have a test for the ssh server feature. but not now.

File: border0/resource_connector_token.go [hunk 3]
```diff
@@ -101,7 +107,7 @@ func resourceConnectorTokenCreate(ctx context.Context, d *schema.ResourceData, m
 		return diagnosticsError(err, "Failed to create connector token")
 	}
 
-	d.SetId(created.ID)
+	d.SetId(connectorID + ":" + created.ID)
```
Comment (adrianosela) [inline border0/resource_connector_token.go:110]: format strings prefered.

File: border0/resource_policy.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+	if err := json.Unmarshal([]byte(old), &oldJSONAsPolicyData); err != nil {
+		return false
+	}
+
+	if err := json.Unmarshal([]byte(new), &newJSONAsPolicyData); err != nil {
+		return false
+	}
+
+	return reflect.DeepEqual(oldJSONAsPolicyData, newJSONAsPolicyData)
```
Comment (adrianosela) [inline border0/resource_policy.go:170]: nice

**Review by adrianosela — APPROVED** [review]
Ship it 🚢

File: border0/resource_connector_test.go (hunk 1 above)
Comment (waltzofpearls) [reply border0/resource_connector_test.go:64]: No read after. Delete (terraform destroy) is done automatically by the terraform unit test function.

File: border0/resource_connector_test.go (hunk 2 above)
Comment (waltzofpearls) [reply border0/resource_connector_test.go:15]: Yeah, agreed.

**Pull request merged**